core/src/test/scala/integration/kafka/api/MetricsTest.scala

```diff
@@ -205,15 +205,13 @@ class MetricsTest extends IntegrationTestHarness with SaslSetup {
   }
 
   private def verifyBrokerZkMetrics(server: KafkaServer, topic: String): Unit = {
-    // Latency is rounded to milliseconds, so we may need to retry some operations to get latency > 0.
-    val (_, recorded) = TestUtils.computeUntilTrue({
-      servers.head.zkClient.getLeaderForPartition(new TopicPartition(topic, 0))
-      yammerMetricValue("kafka.server:type=ZooKeeperClientMetrics,name=ZooKeeperRequestLatencyMs").asInstanceOf[Double]
-    })(latency => latency > 0.0)
-    assertTrue("ZooKeeper latency not recorded", recorded)
-
-    assertEquals(s"Unexpected ZK state ${server.zkUtils.zkConnection.getZookeeperState}",
-        "CONNECTED", yammerMetricValue("SessionState"))
+    // Latency is rounded to milliseconds, so check the count instead.
+    val initialCount = yammerHistogramCount("kafka.server:type=ZooKeeperClientMetrics,name=ZooKeeperRequestLatencyMs").asInstanceOf[Long]
```

Comment:
So, we don't actually need to wait until true here?

Reply:
Right, since the ZK write in between is blocking and is guaranteed to update the count in the histogram.

Reply:
Ah, I had missed that you're using the count now. Makes sense.